src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -652,7 +652,7 @@ public static void Write<T>(void* destination, T value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static ref T AsRef<T>(void* source)
         {
-            return ref Unsafe.As<byte, T>(ref *(byte*)source);
+            return ref *(T*)source;
 
             // ldarg .0
```

Comment:
Comments showing il disassembly and how methods are referenced in different runtimes can be deleted on the methods that you are touching.
```
        // CoreCLR:METHOD__UNSAFE__AS_REF_POINTER
        // AOT:AsRef
        // Mono:AsRef
```